src/codegen/mod.rs

```diff
@@ -2400,28 +2400,33 @@ impl CodeGenerator for Function {
     }
 }
 
-impl CodeGenerator for ObjCInterface {
-    type Extra = Item;
-    fn codegen<'a>(&self,
-                   ctx: &BindgenContext,
-                   result: &mut CodegenResult<'a>,
-                   _whitelisted_items: &ItemSet,
-                   _: &Item) {
-        let mut impl_items = vec![];
-        let mut trait_items = vec![];
 
-        for method in self.methods() {
+fn objc_method_codegen(ctx: &BindgenContext,
+                       method: &ObjCMethod,
+                       class_name: Option<&str>)
+                       -> (ast::ImplItem, ast::TraitItem) {
             let signature = method.signature();
             let fn_args = utils::fnsig_arguments(ctx, signature);
             let fn_ret = utils::fnsig_return_ty(ctx, signature);
-            let sig = aster::AstBuilder::new()
+    let sig;
+
+    if method.is_class() {
+        sig = aster::AstBuilder::new()
+            .method_sig()
+            .unsafe_()
+            .fn_decl()
+            .with_args(fn_args.clone())
+            .build(fn_ret);
+    } else {
+        sig = aster::AstBuilder::new()
                 .method_sig()
                 .unsafe_()
                 .fn_decl()
                 .self_()
                 .build(ast::SelfKind::Value(ast::Mutability::Immutable))
                 .with_args(fn_args.clone())
                 .build(fn_ret);
+    }
 
             // Collect the actual used argument names
             let arg_names: Vec<_> = fn_args.iter()
```

Comment:
Ehh, I guess I had ignore whitespaces on when adding chunks to the commit, indent should be better now